portal/index.php

```diff
@@ -523,6 +520,13 @@ function callServer(action, value, value2, last, first) {
                 alert(message);
             });
         }
+
+    function bsAlert(msg) {
+        let htm = $($("#alertStore").html());
+        $('<strong>'+msg+'</strong>').insertBefore(htm.find("button:first"));
+        htm.insertBefore("form");
+        setTimeout(()=>{$('.alert').alert('close');}, 5000);
```

Comment:
Would be really nice to be able to drop the gritter library.
Are you able to use standard js in above lines 525, 526, 528 rather than jquery (we are trying to use standard js in new code rather than jquery)?

Reply:
Had the developer do this conversion to test productivity impact.

While using standard js is nice objective, project risks introducing unstable code scattered throughout codebase unless a suitable library of jquery like functions or a complete framework is specified.

Reply:
I agree with @bradymiller here, would prefer this be standard JS instead of jQuery. The use of jQuery does not mitigate against unstable code.

Reply:
1. As stated, let's stipulate use standard js is nice objective. Then there is matter of limited money and time. Without any suitable alternative, every dev is going to learn yet another way of making these minor changes on our dime. That leads to ..
2. Completely disagree with stability assessment. Any modular, mature development is more stable than new, ad-hoc block of code. Leading towards obsolescence, yes but not unstable. 
3. To be clear, organizations use external components to leverage (free) fruits of efforts of additional developers with free testing and maintenance to boot. Replacing jQ is much bigger lift that jQUI that is embedded here. A bigger and more organized approach is necessary if that is to happen.

Reply:
Let's be clear, refactoring jquery with javascript will never happen imo.  Besides, what's the big deal with using jquery. It's just JS wrapped code anyway no diff than any other framework. Are we next going to tackle refactoring Symfony functions to pure php because it's not pure enough?

Reply:
The conversation at hand is not whether or not to refactor every line of jQuery in OpenEMR with vanilla javascript. Instead, I think we should strive to continually embrace current best-practices; if we don't take small steps forward, the codebase doesn't progress.

Reply:
Keyword here is 'current'. What is considered best practices today may not be tomorrow and wasn't yesterday perhaps!
Good engineers evaluate and use available tools based on the design goals for a project. Team talent resources play a role helping to determine what is the best practice for a project.

Pure Javascript is most times cumbersome.. This is why we have libraries that wrapper Javascript. While you and I may not have difficulty writing Javascript, there are many that only use jquery.

I thought we decided that if the core code being modified was jquery then we'd continue using jquery for that feature/module. In the end, I agree we should promote Javascript for new code however for consistency, we follow the original design intent when doing modifications.

Reply:
I pointed this out since it was a new block of code, so figured best to go with pure js.

To simplify this discussion.
For completely new code, should we be using jquery or pure js?

Reply:
I would say new code should use pure JS;.